sale_order_secondary_unit/models/sale_order.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2018-2020 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = ["sale.order.line", "product.secondary.unit.mixin"]
+    _name = "sale.order.line"
+    _secondary_unit_fields = {
+        "qty_field": "product_uom_qty",
+        "uom_field": "product_uom",
+    }
+
+    secondary_uom_qty = fields.Float(string="2nd Qty", digits="Product Unit of Measure")
+    secondary_uom_id = fields.Many2one(
+        comodel_name="product.secondary.unit",
+        string="2nd uom",
+        ondelete="restrict",
+    )
+
+    secondary_uom_unit_price = fields.Float(
+        string="2nd unit price",
+        digits="Product Unit of Measure",
+        store=False,
+        readonly=True,
+        compute="_compute_secondary_uom_unit_price",
+    )
+
+    product_uom_qty = fields.Float(
+        store=True, readonly=False, compute="_compute_product_uom_qty", copy=True
+    )
+
+    @api.depends("secondary_uom_qty", "secondary_uom_id", "product_uom_qty")
+    def _compute_product_uom_qty(self):
+        self._compute_helper_target_field_qty()
+
+    @api.onchange("product_uom")
+    def onchange_product_uom_for_secondary(self):
+        self._onchange_helper_product_uom_for_secondary()
+
+    @api.onchange("product_id")
+    def product_id_change(self):
+        """
+        If default sales secondary unit set on product, put on secondary
+        quantity 1 for being the default quantity. We override this method,
+        that is the one that sets by default 1 on the other quantity with that
+        purpose.
+        """
+        res = super().product_id_change()
+        self.secondary_uom_id = self.product_id.sale_secondary_uom_id
+        if self.secondary_uom_id:
+            self.secondary_uom_qty = 1.0
+            self.onchange_product_uom_for_secondary()
+        return res
+
+    @api.depends("secondary_uom_qty", "product_uom_qty", "price_unit")
+    def _compute_secondary_uom_unit_price(self):
+        for line in self:
+            if line.secondary_uom_id:
+                try:
+                    line.secondary_uom_unit_price = (
+                        line.price_subtotal / line.secondary_uom_qty
+                    )
+                except ZeroDivisionError:
+                    line.secondary_uom_unit_price = 0
+            else:
+                line.secondary_uom_unit_price = 0
```

Comment:
```suggestion
            if line.secondary_uom_id and line.secondary_uom_qty:
                line.secondary_uom_unit_price = (
                    line.price_subtotal / line.secondary_uom_qty
                )
            else:
                line.secondary_uom_unit_price = 0
```

Reply:
Travis was already red before, the module sale_order_qty_change_no_recompute is in conflict with the computed product_uom_qty field in this module.

Reply:
As not being a Travis erudit, it looks strange to me.

I ran test cases 1 by 1, (with `setupClass()` between) and re-run tests with `odoo --test-enable` and no error is raised

Too, i tested with odoo standard --test-enable both modules installed both pass without error (i had to change a partner reference for sale_order_qty_change_no_recompute test setup).